src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -245,16 +260,30 @@ CreateReferenceTable(Oid relationId)
 						errdetail("There are no active worker nodes.")));
 	}
 
+	/* we only support data loading for regular (non-foreign) relations */
+	relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_RELATION)
+	{
+		canLoadData = true;
+	}
+
 	colocationId = CreateReferenceTableColocationId();
 
 	/* first, convert the relation into distributed relation */
 	ConvertToDistributedTable(relationId, distributionColumnName,
-							  DISTRIBUTE_BY_NONE, REPLICATION_MODEL_2PC, colocationId);
+							  DISTRIBUTE_BY_NONE, REPLICATION_MODEL_2PC, colocationId,
+							  canLoadData);
```

Comment:
Reusing this variable for the `allowEmpty` parameter was also confusing to me. Can we push that logic into `CopyLocalDataIntoShards`? Maybe call it `CopyLocalRelationDataIntoShards` to distinguish from foreign tables? Dunno.